misc/ls-extensions/modules/metadata-to-record-generator/src/main/java/io/ballerina/generators/platform/SalesforceMetadataConverter.java

```diff
@@ -0,0 +1,150 @@
+import static io.ballerina.compiler.syntax.tree.NodeFactory.createTypeDefinitionNode;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.CLOSE_BRACE_TOKEN;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.EQUAL_TOKEN;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.OPEN_BRACE_TOKEN;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.PUBLIC_KEYWORD;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.QUESTION_MARK_TOKEN;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.RECORD_KEYWORD;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.SEMICOLON_TOKEN;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.TYPE_KEYWORD;
+import static io.ballerina.generators.utils.Constants.SF_WHITESPACE_TYPE;
+import static io.ballerina.generators.utils.ConverterUtils.convertSalesforceTypeToBallerina;
+import static io.ballerina.generators.utils.ConverterUtils.escapeIdentifier;
+
+/**
+ * Converts the JSON object containing metadata of a Salesforce sObject to a Ballerina record.
+ */
+public class SalesforceMetadataConverter implements MetadataConverter {
+    @Override
+    public MetadataToRecordSchemaResponse convert(String metadata) throws MetadataToRecordGeneratorException {
+        try {
+            SalesforceRecord recordData = salesforceRecordExtractor(metadata);
+            String name = ((recordData.getRecordName() != null) && !recordData.getRecordName().equals("")) ?
+                    recordData.getRecordName() : "NewRecord";
+            MetadataToRecordSchemaResponse response = new MetadataToRecordSchemaResponse();
+            NodeList<ImportDeclarationNode> imports = createEmptyNodeList();
+            List<Node> recordFieldList = new ArrayList<>();
+            recordData.getFields().forEach((fieldData) -> {
+                RecordFieldNode recordFieldNode = null;
+                RecordFieldWithDefaultValueNode recordFieldNodeWithDefaultValue = null;
+
+                String dataType = convertSalesforceTypeToBallerina(fieldData.getType());
+                String fieldName = escapeIdentifier(fieldData.getName());
+                boolean isNillable = fieldData.isNillable();
+                if (fieldData.isUpdateable() || fieldData.isCreateable()) {
+                    if (isNillable) {
+                        recordFieldNodeWithDefaultValue = createRecordFieldWithDefaultValueNode(null, null,
+                                createIdentifierToken(dataType +
+                                        createToken(QUESTION_MARK_TOKEN) + SF_WHITESPACE_TYPE),
+                                createIdentifierToken(fieldName),
+                                createToken(EQUAL_TOKEN),
+                                createSimpleNameReferenceNode(createIdentifierToken("()")),
+                                createToken(SEMICOLON_TOKEN));
+                        recordFieldList.add(recordFieldNodeWithDefaultValue);
```

Comment:
Consider a scenario where a Salesforce object is created and will be updated in a next request. The default values may overwrite exisiting values if the user ignores the field.